image_viewer.cpp

```diff
@@ -7,32 +7,33 @@
 #include "sdlutils.h"
 #include "text_viewer.h"
 
-#define VIEWER_MARGIN 1
+#include <algorithm>
+#include <sys/stat.h>
 
 ImageViewer::ImageViewer(CPanel *panel)
-    : panel_(panel)
+    : panel_(panel), showTitle_(false)
 {
     init();
     setPath(panel->getHighlightedItemFull());
 }
 
 void ImageViewer::init()
 {
-    // Create background image
-    background_ = SDLSurfaceUniquePtr { SDL_utils::createImage(screen.actual_w,
-        screen.actual_h, SDL_MapRGB(screen.surface->format, COLOR_BG_1)) };
-
-    // Transparency grid background.
+    // Create the fullscreen background surface
+    background_ = SDLSurfaceUniquePtr{
+        SDL_utils::createImage(screen.actual_w,
+            screen.actual_h, SDL_MapRGB(screen.surface->format, COLOR_BG_1))
+    };
+    // Checkerboard background
     constexpr int kTransparentBgRectSize = 10;
     const Uint32 colors[2] = {
         SDL_MapRGB(background_->format, 240, 240, 240),
         SDL_MapRGB(background_->format, 155, 155, 155),
     };
-    int j = 0;
     const int rect_w = static_cast<int>(kTransparentBgRectSize * screen.ppu_x);
     const int rect_h = static_cast<int>(kTransparentBgRectSize * screen.ppu_y);
-    for (int j = 0, y = Y_LIST * screen.ppu_y; y < screen.actual_h;
-         y += rect_h, ++j) {
+
+    for (int j = 0, y = 0; y < screen.actual_h; y += rect_h, ++j) {
```

Comment:
The initial `y` should probably depend on whether title bar is showing:
```suggestion
    for (int j = 0, y = showTitle_ ? Y_LIST * screen.ppu_y : 0; y < screen.actual_h;
         y += rect_h, ++j) {
```